File: csharp/Validators/AbstractValidator.cs
using System;
using System.Collections.Generic;

namespace Validators
{
    public abstract class AbstractValidator
    {
        protected List<string> Arguments { get; set; }
        public string Message { get; protected set; }

        public AbstractValidator(List<string> args, string message)
        {
            Arguments = args;
            Message = message;
        }

        public abstract bool Validate();
    }
}

File: java/JavaFileutil/src/JavaFileutil/Validators/AbstractValidator.java
package JavaFileutil.Validators;

import java.util.List;

public abstract class AbstractValidator
{
    protected List<String> Arguments;
    protected String Message;

    public AbstractValidator(List<String> args, String message)
    {
        Arguments = args;
        Message = message;
    }

    public abstract boolean Validate();
}
